3p-ecosystem/src/main/java/com/google/homesampleapp/screens/inspect/InspectFragment.kt

```diff
@@ -85,108 +116,156 @@ class InspectFragment : Fragment() {
   }
 
   // -----------------------------------------------------------------------------------------------
-  // Setup Observers
-
-  private fun setupObservers() {
-    // Observer on the currently selected device
-    selectedDeviceViewModel.selectedDeviceIdLiveData.observe(viewLifecycleOwner) { deviceId ->
-      Timber.d(
-          "selectedDeviceViewModel.selectedDeviceIdLiveData.observe is called with deviceId [${deviceId}]")
-      updateDeviceInfo()
-    }
+  // Composables
 
-    // Observer on introspection information
-    viewModel.instrospectionInfo.observe(viewLifecycleOwner) { updateIntrospectionInfo(it) }
-  }
+  @Composable
+  private fun InspectRoute() {
+    // Observes values needed by the InspectScreen.
+    val selectedDeviceId by selectedDeviceViewModel.selectedDeviceIdLiveData.observeAsState()
+    val selectedDevice by selectedDeviceViewModel.selectedDeviceLiveData.observeAsState()
+    val instrospectionInfo by viewModel.instrospectionInfo.observeAsState()
 
-  // -----------------------------------------------------------------------------------------------
-  // UI update functions
+    binding.topAppBar.title = selectedDevice?.device?.name
```

Comment:
Since the `topAppBar` is a View, updating it in the Composition inherently makes the composition unstable. Is it possible to use Compose for the toolbar as well?

If not, updating the title should ideally still be done outside of the Composition. Shortly after `binding` is initialized, the following can be done:

```
binding = ...
selectedDeviceViewModel.selectedDeviceLiveData.observe(viewLifecycleOwner) {
     binding.topAppBar.title = it?.device?.name
}

```

Reply:
I was hoping to get to the toolbar at the last step, when I migrate the headers and navigation to compose.
So for now, I simply applied the fix you suggested.